### PR TITLE
add initial codeowners file

### PR DESCRIPTION
**Pull request by rosskirkpat**
As part of addressing some recent issues surrounding changes made that affected CI and were not caught prior to being merged, I am adding a minimal guardrail using a CODEOWNERS file.

PRs will require review and approval by ~~@oxr463, @paraglade,~~ @superseb, @phillipsj , and myself for any changes made to the Drone yml file.

PRs will require review and approval by @phillipsj and myself for any changes to Windows related files.

PRs will require review and approval by @Oats87 and myself for any changes made to the shell scripts for provisioning tests.

### CODE REVIEW

**Comment by rosskirkpat** [comment]
This should be merged after https://github.com/rancher/rancher/pull/37892

**Review by phillipsj — DISMISSED** [review]

**Comment by superseb** [comment]
Minimal guard rails is different from introducing SPOF, adding single persons will create a blocking problem as soon as that person is not working and in general, keep relying on individuals will not make anything more flexible.

Drone build configuration is something that Engineering owns, that EIO is running the infra does not mean that they need to be introduced to review our configuration. And again, a single individual next to EIO that will review is not flexible.

**Comment by oxr463** [comment]
> Minimal guard rails is different from introducing SPOF, adding single persons will create a blocking problem as soon as that person is not working and in general, keep relying on individuals will not make anything more flexible.

I think the goal here is to get more review on changes that impact the pipeline.

> Drone build configuration is something that Engineering owns, that EIO is running the infra does not mean that they need to be introduced to review our configuration. And again, a single individual next to EIO that will review is not flexible.

Agreed. I don't think EIO is the proper owner of this pipeline. However, I do think it would be a good idea for one of the teams to own this.

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.drone.yml @oxr463 @paraglade @rosskirkpat
```
Comment (oxr463) [inline CODEOWNERS:1]: I don't think this should be EIO.

**Review by oxr463 — CHANGES_REQUESTED** [review]

File: CODEOWNERS (hunk 1 above)
Comment (rosskirkpat) [reply CODEOWNERS:1]: I agree that it should not be wholly owned by EIO, however, at this time I don't think any Engineering team (or sub-set of a team, like what EIO is), has a clear ownership/knowledge of Drone other than EIO. That was why I included both of you. This is something we should address at a higher level of SMEs and technical ownership of a product/project.

I know that @phillipsj and I have spent a lot of time working in Drone across many projects in Rancher. I will change this to include the two of us and @superseb

**Pull request closed without merging**